homeassistant/components/netatmo/cover.py

```diff
@@ -0,0 +1,144 @@
+"""Support for Netatmo/Bubendorff covers."""
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from pyatmo import modules as NaModules
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_URL_CONTROL, NETATMO_CREATE_COVER
+from .data_handler import HOME, SIGNAL_NAME, NetatmoDevice
+from .netatmo_entity_base import NetatmoBase
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Netatmo cover platform."""
+
+    @callback
+    def _create_entity(netatmo_device: NetatmoDevice) -> None:
+        entity = NetatmoCover(netatmo_device)
+        _LOGGER.debug("Adding cover %s", entity)
+        async_add_entities([entity])
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, NETATMO_CREATE_COVER, _create_entity)
+    )
+
+
+class NetatmoCover(NetatmoBase, CoverEntity):
+    """Representation of a Netatmo cover device."""
+
+    def __init__(self, netatmo_device: NetatmoDevice) -> None:
+        """Initialize the Netatmo device."""
+        CoverEntity.__init__(self)
+        super().__init__(netatmo_device.data_handler)
+        self.optimistic = True
+
+        self._cover = cast(NaModules.Shutter, netatmo_device.device)
+
+        self._id = self._cover.entity_id
+        self._attr_name = self._device_name = self._cover.name
+        self._model = self._cover.device_type
+        self._config_url = CONF_URL_CONTROL
+
+        self._home_id = self._cover.home.entity_id
+        self._closed: bool | None = None
+        self._attr_is_closed = None
+
+        self._signal_name = f"{HOME}-{self._home_id}"
+        self._publishers.extend(
+            [
+                {
+                    "name": HOME,
+                    "home_id": self._home_id,
+                    SIGNAL_NAME: self._signal_name,
+                },
+            ]
+        )
+        self._attr_unique_id = f"{self._id}-{self._model}"
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        supported_features = 0
+        supported_features |= CoverEntityFeature.OPEN
+        supported_features |= CoverEntityFeature.CLOSE
+        supported_features |= CoverEntityFeature.STOP
+        supported_features |= CoverEntityFeature.SET_POSITION
+
+        return supported_features
```

Comment:
Static, can be defined globally and assign in class instance

Reply:
This will probably not stay static once different devices will be implemented with different feature sets but it can be pulled out again later.

Reply:
This is the basics right? So it should be static regardless, but adding tilt, etc. would be based on device

Reply:
Exactly. Since none of the testers had such a device and the documentation is lacking details I only went with the very basics.